### DISCUSSION TITLE
Does GetCallerIdentityRequest support getting identity/arn using a given sts token?

### DISCUSSION THREAD

**Discussion by yjanezou** [aws/aws-sdk-java#3013] [Q&A]
We are building a service which needs to identify caller's identity by using caller's sts token. I wonder if the aws sdk support it?  sample code is as below:

private static final String X_AMZ_DATE_HEADER = "X-Amz-Date";
private static final String X_AMZ_SECURITY_TOKEN_HEADER = "X-Amz-Security-Token";
private static final String AUTHORIZATION_HEADER = "Authorization";

...
AWSSecurityTokenService stsClient = ....;
...

GetCallerIdentityRequest request = new GetCallerIdentityRequest();

request.putCustomRequestHeader(AUTHORIZATION_HEADER, "AWS4-HMAC-SHA256");
request.putCustomRequestHeader(X_AMZ_DATE_HEADER, "20230727T182602Z");
request.putCustomRequestHeader(X_AMZ_SECURITY_TOKEN_HEADER, "IQoJb3JpZ2luX2VjEJn....");

GetCallerIdentityResult getCallerIdentityResult = stsClient.getCallerIdentity(request);

**Comment by yasminetalby** [comment]
Hello @yjanezou ,

Thank you very much for your submission.

Both the AWS JAVA SDK V1 and V2 offer support for getCallerIdentity:
- AWS JAVA SDK V1 : [getCallerIdentity(GetCallerIdentityRequest getCallerIdentityRequest)](https://docs.aws.amazon.com/AWSJavaSDK/latest/javadoc/com/amazonaws/services/securitytoken/AWSSecurityTokenService.html#getCallerIdentity-com.amazonaws.services.securitytoken.model.GetCallerIdentityRequest-)
- AWS JAVA SDK V2 : [getCallerIdentity(GetCallerIdentityRequestgetCallerIdentityRequest)](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/sts/StsClient.html#getCallerIdentity(software.amazon.awssdk.services.sts.model.GetCallerIdentityRequest))

I see that you are already using this in your sample, are you encountering any issue?

Best regards,

Yasmine
